run.cc

```diff
@@ -60,20 +65,240 @@ void runTests() {
+    Data file(Datenliste[i]);
+    Dataset.push_back(file);
+    }
+
+
+    cout << "******************************************************" << endl;
+    runTests();
+    cout << "******************************************************" << endl;
+    // create an object which holds data of experiment A
+
+    Data datA("exp_A");
+    Data datB("exp_B");
+    Data datC("exp_C");
+    Data datD("exp_D");
+
+    //Data& rdatA = datA;
+    //Data& rdatB = datB;
+
+
+    // here is the data from experiment A
+    cout << "bin 27: from " << datA.binLow(27) << " to " << datA.binHigh(27)
+         << endl;
+    cout << "measurement of experiment A in bin 27: " << datA.measurement(27)
+         << endl;
+    cout << "measurement of experiment A in bin 27: " << datA.error(27)
+        << endl;
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 1c)" << endl;
+
+    cout << "measurement of experiment A in bin 27:" << Dataset[0].measurement(27) << endl;
+    cout << "measurement of experiment B in bin 27:" << Dataset[1].measurement(27) << endl;
+    cout << "measurement of experiment C in bin 27:" << Dataset[2].measurement(27) << endl;
+    cout << "measurement of experiment D in bin 27:" << Dataset[3].measurement(27) << endl;
+
+
+    double a = Dataset[0].measurement(27) - Dataset[1].measurement(27);
+    double b = 2*sqrt(pow(Dataset[0].error(27), 2) + pow(Dataset[1].error(27), 2));
+    double c = b- a;
+
+    cout << "bin 27, c: " << c << endl;
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 1d)" << endl;
+
+    //checkCompatibility(datA, datD, 2);
+    cout << "DatA/DatB" << endl;
+    datA.checkCompatibility(datB, 2);
+    cout << "DatA/DatC" << endl;
+    datA.checkCompatibility(datC, 2);
+    cout << "DatA/DatD" << endl;
+    datA.checkCompatibility(datD, 2);
+    cout << "DatB/DatC" << endl;
+    datB.checkCompatibility(datC, 2);
+    cout << "DatB/DatD" << endl;
+    datB.checkCompatibility(datD, 2);
+
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 1e)" << endl;
+    //cout << "DatA/DatD" << endl;
+
+    Data datav = average(datA,datB);
+    cout << "measurement of average of a and b in bin 27: " << datav.measurement(27)
        << endl;
 
-  return 0;
+    //average(datA, datD);
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 2b)" << endl;
+
+    datA.x2test();
+    datB.x2test();
+    datC.x2test();
+    datD.x2test();
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 2a)" << endl;
+
+    datA.check(2);
+    datB.check(2);
+    datC.check(2);
+    datD.check(2);
+
+    datA.check(3);
+    datB.check(3);
+    datC.check(3);
+    datD.check(3);
+
+    cout << "******************************************************" << endl;
+    cout << "Aufgabe 2c)" << endl;
+
+    x2Compare();
+
```

Comment:
You didn't answer if the data are compatible with the model using the given parameters. The chisquare values you printed out are correct, but do you think they are 'good'? What should the value ideally be and are the deviations from that value that you see here significant? -0.5p